service/notesservice.php

```diff
@@ -196,6 +181,33 @@ public function delete ($id, $userId) {
         $file->delete();
     }
 
+    private function getSafeTitleFromContent($content) {
+        // prepare content: remove markdown characters and empty spaces
+        $content = preg_replace("/^\s*[*+-]\s+/m", "", $content); // list item
+        $content = preg_replace("/^#+\s+(.*?)\s*#*$/m", "$1", $content); // headline
+        $content = preg_replace("/^(?:=*|-*)$/m", "", $content); // separate line for headline
+        $content = preg_replace("/(\*+|_+)(.*?)\\1/m", "$2", $content); // emphasis
+        $content = preg_replace("/^\s+/m", "", $content); // whitespace at beginning
+        $content = preg_replace("/\s+$/m", "", $content); // whitespace at end
+        $content = trim($content);
+
+        // generate content from the first line of the title
+        $splitContent = preg_split("/\R/", $content, 2);
+        $title = $splitContent[0];
```

Comment:
The advantage of doing the replacements first is that removing markups can produce an empty line which would result in an empty title. For example, if the first line consists of some characters that form a markdown code that is removed by the regexps, only some whitespaces are left. I wanted to prevent this, since an empty title is useless.